test/endpoint_provisioner_test.go

```diff
@@ -0,0 +1,1083 @@
+			clientResponse, _, err := testcontext.SEMP().Monitor().MsgVpnApi.
+				GetMsgVpnQueue(testcontext.SEMP().MonitorCtx(), testcontext.Messaging().VPN, url.QueryEscape(provisionQueueName), nil)
+			Expect(err).ToNot(BeNil())
+
+			// MaxRedeliveryCount(0) - Now let's provision the endpoint using the API
+			provisioner = provisioner.FromConfigurationProvider(config.EndpointPropertyMap{
+				config.EndpointPropertyDurable:              true,
+				config.EndpointPropertyMaxMessageRedelivery: 0,
+			})
+			outcome := provisioner.Provision(provisionQueueName, false)
+			Expect(outcome.GetStatus()).To(BeTrue())
+			Expect(outcome.GetError()).ToNot(HaveOccurred())
+
+			// check that the endpoint was provisioned on the broker via semp
+			clientResponse, _, err = testcontext.SEMP().Monitor().MsgVpnApi.
+				GetMsgVpnQueue(testcontext.SEMP().MonitorCtx(), testcontext.Messaging().VPN, url.QueryEscape(provisionQueueName), nil)
+			Expect(err).To(BeNil())
+			Expect(clientResponse.Data).ToNot(BeNil())
+			Expect(clientResponse.Data.QueueName).To(Equal(provisionQueueName))
+			Expect(clientResponse.Data.Durable).ToNot(BeNil())   // durability should not be nil
+			Expect(**(clientResponse.Data.Durable)).To(BeTrue()) // durability should be true
+			Expect(clientResponse.Data.MaxRedeliveryCount).To(Equal(int64(0)))
+			// remove the queue from the broker
+			err = provisioner.Deprovision(provisionQueueName, true)
+			Expect(err).To(BeNil())
+
+			// MaxRedeliveryCount(255) - Now let's provision the endpoint using the API
+			provisioner = provisioner.FromConfigurationProvider(config.EndpointPropertyMap{
+				config.EndpointPropertyDurable:              true,
+				config.EndpointPropertyMaxMessageRedelivery: 255,
+			})
+			outcome = provisioner.Provision(provisionQueueName, false)
+			Expect(outcome.GetStatus()).To(BeTrue())
+			Expect(outcome.GetError()).ToNot(HaveOccurred())
+
+			// check that the endpoint was provisioned on the broker via semp
+			clientResponse, _, err = testcontext.SEMP().Monitor().MsgVpnApi.
+				GetMsgVpnQueue(testcontext.SEMP().MonitorCtx(), testcontext.Messaging().VPN, url.QueryEscape(provisionQueueName), nil)
+			Expect(err).To(BeNil())
+			Expect(clientResponse.Data).ToNot(BeNil())
+			Expect(clientResponse.Data.QueueName).To(Equal(provisionQueueName))
+			Expect(clientResponse.Data.Durable).ToNot(BeNil())   // durability should not be nil
+			Expect(**(clientResponse.Data.Durable)).To(BeTrue()) // durability should be true
+			Expect(clientResponse.Data.MaxRedeliveryCount).To(Equal(int64(255)))
+			// remove the queue from the broker
+			err = provisioner.Deprovision(provisionQueueName, true)
+			Expect(err).To(BeNil())
+
+			// MaxRedeliveryCount(-1) - Now let's provision the endpoint using the API
+			provisioner = provisioner.FromConfigurationProvider(config.EndpointPropertyMap{
+				config.EndpointPropertyDurable:              true,
+				config.EndpointPropertyMaxMessageRedelivery: -1,
```

Comment:
`-1` and `256` are already tested at the top of this file, is this testing a different code path by not using the builder methods?

Reply:
But is it different than the previous tests with the same values, no sense in testing it twice if we don't need to